src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/MultipartFormDataRequestContentProvider.cs

```diff
@@ -0,0 +1,495 @@
+    {
+        private static readonly Lazy<MultipartFormDataRequestContentProvider> _instance = new(() => new MultipartFormDataRequestContentProvider());
+        private class FormDataItemTemplate<T> { }
+        public static MultipartFormDataRequestContentProvider Instance => _instance.Value;
+        private MultipartFormDataRequestContentProvider() : base(Configuration.HelperNamespace, null)
+        {
+            DeclarationModifiers = TypeSignatureModifiers.Internal;
+            Inherits = Configuration.ApiTypes.RequestContentType;
+            _multipartContentField = new FieldDeclaration(
+                modifiers: FieldModifiers.Private | FieldModifiers.ReadOnly,
+                type: typeof(MultipartFormDataContent),
+                name: "_multipartContent");
+            _randomField = new FieldDeclaration(
+                modifiers: FieldModifiers.Private | FieldModifiers.Static | FieldModifiers.ReadOnly,
+                type: typeof(Random),
+                name: "_random",
+                initializationValue: New.Instance(typeof(Random), Array.Empty<ValueExpression>()),
+                serializationFormat: SerializationFormat.Default);
+            _boundaryValuesFields = new FieldDeclaration(
+                modifiers: FieldModifiers.Private | FieldModifiers.Static | FieldModifiers.ReadOnly,
+                type: typeof(char[]),
+                name: "_boundaryValues",
+                initializationValue: Literal("0123456789=ABCDEFGHIJKLMNOPQRSTUVWXYZ_abcdefghijklmnopqrstuvwxyz").ToCharArray(),
+                serializationFormat: SerializationFormat.Default);
+            ValueExpression contentTypeHeaderExpression = new UnaryOperatorExpression("!", new MultipartFormDataContentExpression(_multipartContentField).Property("Headers").Property("ContentType"), true);
+            ValueExpression contentTypeHeaderValueExpression = new InvokeInstanceMethodExpression(contentTypeHeaderExpression, nameof(MediaTypeHeaderValue.ToString), Array.Empty<ValueExpression>(), null, false);
+            _contentTypeProperty = new PropertyDeclaration(
+                description: null,
+                modifiers: MethodSignatureModifiers.Public,
+                propertyType: typeof(string),
+                name: _contentTypePropertyName,
+                propertyBody: new MethodPropertyBody(new MethodBodyStatement[]
+                {
+                    Return(contentTypeHeaderValueExpression)
+                }));
+            _httpContentProperty = new PropertyDeclaration(
+                description: null,
+                modifiers: MethodSignatureModifiers.Internal,
+                propertyType: typeof(HttpContent),
+                name: _httpContentPropertyName,
+                propertyBody: new ExpressionPropertyBody(_multipartContentField)
+                );
+            _multipartContentExpression = new MultipartFormDataContentExpression(_multipartContentField);
```

Comment:
move this up so that we could use this in `contentTypeHeaderValueExpression` (L53)